### PR TITLE
feat(localization): add German localization

### PR DESCRIPTION
**Pull request by mkoehne**
# Submit a pull request

## CLA

- [X] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [X] The code changes follow best practices
- [X] Code changes are tested (add some information if not applicable)

## Description of the pull request
Added German localization.

### CODE REVIEW

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+
+  @override
+  String get replyToMessageLabel => 'Nachricht antworten';
+
+  @override
+  String attachmentLimitExceedError(int limit) =>
+      'Dateigröße überschritten, Grenze: $limit';
+
+  @override
+  String get slowModeOnLabel => 'Langsam-Modus EIN';
```
Comment (DaemonLoki) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart:371]:
```suggestion
  String get slowModeOnLabel => 'Langsamer Modus: EIN';
```

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+  @override
+  String get linkDisabledDetails =>
+      'Das Senden von Links ist in dieser Konversation nicht erlaubt.';
+
+  @override
+  String get linkDisabledError => 'Verknüpfungen sind deaktiviert';
+
+  @override
+  String get sendMessagePermissionError =>
+      'Sie sind nicht berechtigt, Nachrichten zu senden';
```
Comment (DaemonLoki) [inline packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart:382]:
```suggestion
      'Sie sind nicht berechtigt Nachrichten zu senden';
```

**Review by DaemonLoki — COMMENTED** [review]
Added almost all of my corrections as suggestions so that you can quickly use them. :) 

Wonderful day!
Oh, and what do you call a sorceress in the desert? ....a sandwitch

File: packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+  String membersCountText(int count) {
+    if (count == 1) return '1 Mitglied';
+    return '$count Mitglieder';
+  }
+
+  @override
+  String watchersCountText(int count) {
+    if (count == 1) return '1 Online';
+    return '$count Online';
+  }
```
[the remark this answers is not in the document]
Comment (imtoori) [reply packages/stream_chat_localizations/lib/src/stream_chat_localizations_de.dart:305]:
```suggestion
  String watchersCountText(int count) {
    return '$count Online';
  }
```

**Pull request merged**